crates/openvino/src/layout.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{drop_using_function, try_unsafe, util::Result};
+use openvino_sys::{ov_layout_create, ov_layout_free, ov_layout_t};
+use std::ffi::CString;
+
+/// Represents a layout.
+pub struct Layout {
+    instance: *mut ov_layout_t,
+}
+drop_using_function!(Layout, ov_layout_free);
+
+impl Layout {
+    /// Get [`ov_layout_t`] instance.
+    pub fn instance(&self) -> Result<*mut ov_layout_t> {
```

Comment:
This should be `pub(crate)` to avoid leaking the underlying C types:

```suggestion
    pub(crate) fn instance(&self) -> Result<*mut ov_layout_t> {
```